ext/standard/filestat.c

```diff
@@ -955,20 +959,10 @@ PHPAPI void php_stat(const char *filename, size_t filename_length, int type, zva
 		zend_hash_next_index_insert(Z_ARRVAL_P(return_value), &stat_blksize);
 		zend_hash_next_index_insert(Z_ARRVAL_P(return_value), &stat_blocks);
```

Comment:
I tried to put these repetitions inside the array but it didn't work:

```c
		for (int i = 0; i < 13; i++) {
            /* Store numeric indexes in proper order */
            zend_hash_next_index_insert(Z_ARRVAL_P(return_value), stat_sb_addresses[i]);

			/* Store string indexes referencing the same zval*/
			zend_hash_str_add_new(Z_ARRVAL_P(return_value), stat_sb_names[i], strlen(stat_sb_names[i]), stat_sb_addresses[i]);
		}
```

Am I missing something? :)

Reply:
You need to use two separate loops.